plugin/clang.vim

```diff
@@ -840,10 +840,27 @@ func! s:ClangCompleteInit(force)
 endf
 "}}}
 "{{{ ClangExecuteNeoJobHandler
-"handles event: exit
+"Handles events: stdout, stderr, exit
 func! ClangExecuteNeoJobHandler(job_id, data, event)
-  if a:event == 'exit'
-    call ClangExecuteDone(self.fstdout, self.fstderr)
+  if a:event == 'stdout' || a:event == 'stderr'
+    if len(b:clang_state[a:event]) != 0
+      if b:clang_state[a:event][-1] == ''
+        call remove(b:clang_state[a:event], -1)
+      else
+        let line = a:data[0]
+        let b:clang_state[a:event][-1] .= line
+        call remove(a:data, 0)
+      endif
+    endif
```

Comment:
@justmao945 
Update:
I explain it in wrong way. :disappointed: 
> I mean neovim job control will adds a new line to every chunk.

Neovim's job control **only** adds a new line to chunks that not truncated by itself.